### PR TITLE
Fix/credentials conformance

### PR DESCRIPTION
**Pull request by mkhraisha**

### CODE REVIEW

File: src/credentials/dto/credentialOptions.dto.ts [hunk 1]
```diff
@@ -45,14 +54,15 @@ export class CredentialOptionsDTO {
   @IsString()
   domain?: string
 
-  @ApiPropertyOptional({ type: () => CredentialStatusDTO })
-  @IsOptional()
+  @ApiProperty({ type: () => CredentialStatusDTO })
+  @ValidateIf((object, value) => value !== undefined)
```
Comment (vivienfan) [inline src/credentials/dto/credentialOptions.dto.ts:58]: same as above

theoritically, `@IsOptional()` vs `@ValidateIf( !undefined )` should do the same thing.

why the change?
is `@IsOptional` not working?
should we have a ticket to place it everywhere?

**Review by vivienfan — CHANGES_REQUESTED** [review]
some lint issues in `src/general/dto/place.dto.ts`

File: src/credentials/dto/credential.dto.ts [hunk 2]
```diff
@@ -23,23 +22,23 @@ export class CredentialDTO {
   '@context': string[]
 
   @ApiProperty()
-  @IsNotEmpty()
   @IsString()
+  @ValidateIf((object, value) => value !== undefined)
```
[the remark this answers is not in the document]
Comment (mkhraisha) [reply src/credentials/dto/credential.dto.ts:26]: Because this should be "optional" but not allow null

File: src/credentials/dto/credentialOptions.dto.ts (hunk 1 above)
Comment (mkhraisha) [reply src/credentials/dto/credentialOptions.dto.ts:58]: The difference is with regards to how we handle `null` values, this way, nulls are rejected.

File: src/credentials/dto/credential.dto.ts [hunk 3]
```diff
@@ -48,11 +47,6 @@ export class CredentialDTO {
...
   @ApiProperty()
   @IsNotEmpty()
+  @IsObject()
   credentialSubject: unknown
-
-  @ApiPropertyOptional({ type: () => CredentialStatusDTO })
-  @IsOptional()
-  @ValidateNested()
-  @Type(() => CredentialStatusDTO)
-  credentialStatus?: CredentialStatusDTO
```
[the remark this answers is not in the document]
Comment (mkhraisha) [reply src/credentials/dto/credential.dto.ts:57]: There is a distinction between "credential" and "verifiableCredential",
Credential is the unsigned object that we wish to create, and a credentialStatus does not make sense on it. 
verifiableCredential is the signed object that may or may not have credentialStatus

**Review by vivienfan — DISMISSED** [review]
lgtm 👍🏼 , pls fix lint before merge
